libc/math/common.c

```diff
@@ -0,0 +1,109 @@
+/*
+ * Phoenix-RTOS
+ *
+ * libc-tests
+ *
+ * Math tests common part.
+ *
+ * Copyright 2023 Phoenix Systems
+ * Author: Adam Debek
+ *
+ * This file is part of Phoenix-RTOS.
+ *
+ * %LICENSE%
+ */
+
+#include "common.h"
+
+#include <stdlib.h>
+#include <time.h>
+#include <math.h>
+#include <float.h>
+
+double test_getRandomLog(double min, double max)
+{
+	if (min == 0.0) {
+		min = DBL_MIN;
+	}
+	double a = log(min);
+	double b = log(max);
+	double r = (double)rand() / (double)RAND_MAX;
+
+	return exp(a + r * (b - a));
+}
+
+
+double test_getRandomLogSubnrm()
+{
+	double a = log(5.0e-324);
+	double b = log(2.0e-308);
+	double r = (double)rand() / (double)RAND_MAX;
+
+	return exp(a + r * (b - a));
+}
+
+
+void test_setup(void)
+{
+	double num = M_PI;
+	double one = 1.0;
+	double zero = 0.0;
+
+	test_common.maxPowTwoPi = 0;
+	test_common.maxPowTwo = 0;
+	test_common.dblSignif = 0;
+
+	/* Initialize seed for test */
+	srand(time(0));
```

Comment:
I opt for static rand seed too

Reply:
In case of fail, name of function with arguments it was invoked are printed in error message, so reproduction is pretty easy.
Thanks to randomness wider range of arguments is tested.